### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by sumeetjain** [coolingcza/dining_club_web#1]
This is looking _great_ so far. The biggest piece of feedback I have for you is that "helpers" should be used for storing operations that relate specifically to the appearance of the interface. You have helper methods that insert into the database directly. That kind of behavior typically belongs in the related class.

**Comment by coolingcza** [comment]
Could you clarify this a bit?  How/where should my database interaction methods be called if not in main or the helper module?

**Comment by sumeetjain** [comment]
Database interaction methods can be called in main.rb. The methods should be defined in the class or database methods module, called from your "controller" (main.rb), and the resulting appearance should be handled by the views. If needed, helpers are for writing methods that aid in DRYing up the code we use in the views.

**Comment by sumeetjain** [comment]
Let me know if that's not clear. Also, I'd recommend asking the class for examples of their helpers.

**Comment by coolingcza** [comment]
That makes sense...(I think)...
I'm currently using the helper module as a place to store code that would've made the routes much 'thicker'.  Should I replace it in main.rb and reserve the helper module for multiple-use methods?

**Comment by coolingcza** [comment]
Also, could we revisit database interaction?  You mentioned that my helper module is interacting with the database directly, but I'm not sure that I see where.  Is there a line number that you could point to as an example?  Thanks!

**Comment by sumeetjain** [comment]
> I'm currently using the helper module as a place to store code that would've made the routes much 'thicker'. Should I replace it in main.rb and reserve the helper module for multiple-use methods?

Yes. Specifically, save the helper method for things that are related only to appearance. E.g. adding spacing to a string to pad it to 30 characters for better formatting.

Think of helpers specifically as helpers _for the views_. If code wouldn't have gone in a view in the first place, it shouldn't go in a helper.

It's true that helpers can be used in route handlers, but for simplicity's sake I think it's better to think of them as intended for use in views.

---

Example of helper interacting with the database:

https://github.com/coolingcza/dining_club_web/blob/master/helpers/display_logic.rb#L27

(You're right to be confused, because I said "directly" which isn't true here. I meant that you're calling a method for interacting with the database in a helper, which should be avoided.)
